### PR TITLE
WEB-32462 add max_price as input parameter and test

### PR DESCRIPTION
**Pull request by kukla1234**
### What
* add max_price as optional parameter of HotelCollectionRequest
* update documentation to include new parameter
* update HotelCollectionRequest unit test to include new parameter

### How to test
* `composer test-quick` to run unit tests
* changing `max_price` parameter should filter the search results
* need to test it against TAS priceRange ticket

### CODE REVIEW

**Comment by Exeu** [comment]
Within this one you also have to add a function which returns the resultInfo on a hotelcollection result. 
This was added with this feature.

File: src/Request/HotelCollectionRequest.php [hunk 1]
```diff
@@ -92,14 +98,15 @@ class HotelCollectionRequest extends Request
...
-        'roomType' => 'room_type',
+        'item'      => 'item',
+        'path'      => 'path',
+        'currency'  => 'currency',
+        'limit'     => 'limit',
+        'offset'    => 'offset',
+        'order'     => 'order',
+        'category'  => 'category',
+        'roomType'  => 'room_type',
+        'maxPrice' => 'max_price',
```
Comment (Exeu) [inline src/Request/HotelCollectionRequest.php]: alignment here missing one or two spaces

**Comment by kukla1234** [comment]
Added getResultsInfo() and updated tests, docs and test fixtures accordingly.

File: src/Response/HotelCollection/HotelCollection.php [hunk 2]
```diff
@@ -35,6 +35,11 @@ class HotelCollection implements \Countable, \Iterator
     private $searchParams = [];
 
     /**
+     * @var array
```
Comment (Kingson-de) [inline src/Response/HotelCollection/HotelCollection.php]: please provide a class here with some getter functions, in the same style as the other properties (e.g. hotels)

File: src/Response/HotelCollection/HotelCollection.php [hunk 3]
```diff
@@ -122,6 +128,14 @@ public function getSearchParams()
     }
 
     /**
+     * @return array
```
Comment (Exeu) [inline src/Response/HotelCollection/HotelCollection.php]: This needs to be return ResultInfo

**Comment by Kingson-de** [comment]
+1

**Pull request merged**